src/System.Time/src/System/Date.cs

```diff
@@ -0,0 +1,1629 @@
+        /// recognize that the <see cref="DateTime"/> produced by <c>Date.Today.At(TimeOfDay.Now)</c> will have
+        /// <see cref="DateTimeKind.Unspecified"/>, rather than then <see cref="DateTimeKind.Local"/> that would be
+        /// given by <c>DateTime.Now</c>.
+        /// <para>The same applies for <see cref="DateTimeKind.Utc"/>.</para>
+        /// </remarks>
+        public DateTime At(TimeOfDay time)
+        {
+            long ticks = _dayNumber * TimeSpan.TicksPerDay + time.Ticks;
+            return new DateTime(ticks);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="DateTime"/> object from the current <see cref="Date"/>, with the time set to midnight
+        /// (00:00:00). The resulting value has a <see cref="DateTime.Kind"/> of <see cref="DateTimeKind.Unspecified"/>.
+        /// </summary>
+        /// <remarks>
+        /// Note that midnight might be ambiguous or invalid in some time zones on DST transition days.
+        /// Though this method is time zone ignorant, the resulting value should be considered suspect and used with
+        /// caution.
+        /// </remarks>
+        public DateTime ToDateTimeAtMidnight()
+        {
+            return new DateTime(_dayNumber * TimeSpan.TicksPerDay);
+        }
+
+        /// <summary>
+        /// Returns an indication whether the specified year is a leap year.
+        /// </summary>
+        /// <param name="year">A 4-digit year.</param>
+        /// <returns><c>true</c> if year is a leap year; otherwise, <c>false</c>.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="year"/> is less than 1 or greater than 9999.
+        /// </exception>
+        public static bool IsLeapYear(int year)
+        {
+            if (year < 1 || year > 9999)
+                throw new ArgumentOutOfRangeException("year");
+            Contract.EndContractBlock();
+
+            return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
+        }
+
+        /// <summary>
+        /// Returns the number of days in the specified month and year.
+        /// </summary>
+        /// <returns>
+        /// The number of days in <paramref name="month"/> for the specified <paramref name="year"/>.
+        /// For example, if <paramref name="month"/> equals 2 for February, the return value is 28 or 29 depending
+        /// upon whether <paramref name="year"/> is a leap year.
+        /// </returns>
+        /// <param name="year">The year.</param>
+        /// <param name="month">The month (a number ranging from 1 to 12).</param>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="month"/> is less than 1 or greater than 12.
+        /// <para>-or-</para>
+        /// <paramref name="year"/> is less than 1 or greater than 9999.
+        /// </exception>
+        public static int DaysInMonth(int year, int month)
```

Comment:
Yes, and exists as `System.Globalization.Calendar.GetDaysInMonth`.  However there is also `System.DateTime.DaysInMonth`, and since other properties and methods of `Date` are Gregorian-centric (such as `IsLeapYear`, then I think it's ok to have this one here as well.